### DISCUSSION TITLE
Pull action results in staged files

### DISCUSSION THREAD

**Discussion by GShepherdTC** [gitkraken/vscode-gitlens#1254] [Q&A]
Hopefully, this is just a quick setting that I am missing somewhere. 

When I do a pull action after upgrading to the latest version, all of the updated files get pulled down... but also get put into a git stage folder with the current file.  This happens even if I have not changed any local files. 

Does not happen if I use the built in pull functionality of VSC and it didn't do this before the update. 

Any idea why?

**Comment by eamodio** [comment]
I'm not sure what you mean by
>also get put into a git stage folder with the current file

Can you show a screenshot?

**Reply by GShepherdTC** [reply]
Thats going to be hard as I can't post anything real per privacy concerns.  

What happens is when I click the down arrow within gitlens under the COMMITS section for any project, VSC then asked to confirm:
![image](https://user-images.githubusercontent.com/19271498/101925109-7db84800-3b9f-11eb-9b37-4628f6d81ce1.png)
Like it always did before.  

Once confirmed, in addition to all the files it pulls down (remember, I have not touched anything local.. so no merge is required), it puts all the files pulled down into Stage Changes status:

![image](https://user-images.githubusercontent.com/19271498/101925836-5615af80-3ba0-11eb-947f-94314a9b1058.png)

**Reply by GShepherdTC** [reply]
Just checking in to see if anyone has a suggestion?

**Reply by eamodio** [reply]
Sadly, I don't and I can't see how it is related to GitLens (as GitLens just delegates to VS Code for pulling). I believe I have seen this happen myself, but I could never reproduce it. But you see it every time? Do you see it if you pull from the command line? Do you see it if you pull from VS Code (_Git: Pull_ from the command palette)?

**Reply by GShepherdTC** [reply]
Thanks for the feedback.  If I ever figure it out, will write a comment here.  And yes command line git pull works just fine as well as clicking the little 'sync' button in the VSC UI.   Have not tried the command palette version but will do on the next opportunity.  But I suspect that will work too.

**Reply by GShepherdTC** [reply]
I think I figured it out.  There is an option in VSC if you go to settings and search for: autostash  Either I had this checked before I updated to the latest and it didn't cause any auto-staging with GetLens or after I updated, somehow this became checked.  Once I unchecked it, I was able to pull using the Gitlens UI and the changes were not stashed.
